src/python/pants/backend/python/rules/pex.py

```diff
@@ -1,43 +1,48 @@
 # Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
+from dataclasses import dataclass
+from typing import Optional, Tuple
+
 from pants.backend.python.rules.download_pex_bin import DownloadedPexBin
-from pants.backend.python.rules.hermetic_pex import HermeticPex
 from pants.backend.python.subsystems.python_native_code import PexBuildEnvironment
 from pants.backend.python.subsystems.python_setup import PythonSetup
 from pants.backend.python.subsystems.subprocess_environment import SubprocessEncodingEnvironment
-from pants.engine.fs import Digest
+from pants.engine.fs import (EMPTY_DIRECTORY_DIGEST, Digest, DirectoriesToMerge,
+                             DirectoryWithPrefixToAdd)
 from pants.engine.isolated_process import ExecuteProcessResult, MultiPlatformExecuteProcessRequest
 from pants.engine.platform import Platform, PlatformConstraint
 from pants.engine.rules import optionable_rule, rule
 from pants.engine.selectors import Get
-from pants.util.objects import datatype, hashable_string_list, string_optional, string_type
 
 
-class RequirementsPexRequest(datatype([
-  ('output_filename', string_type),
-  ('requirements', hashable_string_list),
-  ('interpreter_constraints', hashable_string_list),
-  ('entry_point', string_optional),
-])):
-  pass
+@dataclass(frozen=True)
+class CreatePex:
+  """Represents a generic request to create a PEX from its inputs."""
+  output_filename: str
+  requirements: Tuple[str] = ()
+  interpreter_constraints: Tuple[str] = ()
+  entry_point: Optional[str] = None
+  input_files_digest: Optional[Digest] = None
 
 
-class RequirementsPex(HermeticPex, datatype([('directory_digest', Digest)])):
-  pass
+@dataclass(frozen=True)
+class Pex:
```

Comment:
This needs to mix in HermeticPex to work. Things _should_ be broken.

Reply:
Sorry, what type of breakage are you referring to?